pkg/gitops/flux/upgrader.go

```diff
@@ -7,22 +7,24 @@ import (
 	"github.com/aws/eks-anywhere/pkg/cluster"
 	"github.com/aws/eks-anywhere/pkg/logger"
 	"github.com/aws/eks-anywhere/pkg/types"
+	releasev1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
 )
 
 const upgradeFluxconfigCommitMessage = "Upgrade commit of flux configuration; generated by EKS-A CLI"
 
-func (f *Flux) Upgrade(ctx context.Context, managementCluster *types.Cluster, currentSpec *cluster.Spec, newSpec *cluster.Spec) (*types.ChangeDiff, error) {
+// Upgrade checks whether upgrading Flux components is necessary and, if so, upgrades to them to the new versions.
+func (f *Flux) Upgrade(ctx context.Context, managementCluster *types.Cluster, currentManagementComponentsVersionsBundle *releasev1.VersionsBundle, currentSpec *cluster.Spec, newSpec *cluster.Spec) (*types.ChangeDiff, error) {
 	logger.V(1).Info("Checking for Flux upgrades")
 
-	changeDiff := FluxChangeDiff(currentSpec, newSpec)
+	changeDiff := ChangeDiff(currentManagementComponentsVersionsBundle, &newSpec.Bundles.Spec.VersionsBundles[0], currentSpec, newSpec)
 	if changeDiff == nil {
 		logger.V(1).Info("Nothing to upgrade for Flux")
 		return nil, nil
 	}
 
 	logger.V(1).Info("Starting Flux upgrades")
 	if err := f.upgradeFilesAndCommit(ctx, newSpec); err != nil {
-		return nil, fmt.Errorf("upgrading Flux from bundles %d to bundles %d: %v", currentSpec.Bundles.Spec.Number, newSpec.Bundles.Spec.Number, err)
+		return nil, fmt.Errorf("upgrading Flux from EKS-A version %s to EKS-A version %s: %v", currentSpec.Cluster.GetManagementComponentsVersion(), *newSpec.Cluster.Spec.EksaVersion, err)
```

Comment:
wouldnt it make more sense to only reference the flux version here?
you can add context about the eksa version in the caller where that context is relevant